lib/galaxy/schema/visualization.py

```diff
@@ -0,0 +1,100 @@
+from datetime import datetime
+from typing import (
+    List,
+    Optional,
+)
+
+from pydantic import (
+    Extra,
+    Field,
+)
+from typing_extensions import Literal
+
+from galaxy.schema.fields import DecodedDatabaseIdField
+from galaxy.schema.schema import (
+    CreateTimeField,
+    Model,
+    TagCollection,
+    UpdateTimeField,
+)
+
+VisualizationSortByEnum = Literal["create_time", "title", "update_time", "username"]
+
+
+class VisualizationIndexQueryPayload(Model):
+    deleted: bool = False
+    show_own: Optional[bool] = None
+    show_published: Optional[bool] = None
+    show_shared: Optional[bool] = None
+    user_id: Optional[DecodedDatabaseIdField] = None
+    sort_by: VisualizationSortByEnum = Field(
+        "update_time", title="Sort By", description="Sort pages by this attribute."
+    )
+    sort_desc: Optional[bool] = Field(default=True, title="Sort descending", description="Sort in descending order.")
+    search: Optional[str] = Field(default=None, title="Filter text", description="Freetext to search.")
+    limit: Optional[int] = Field(default=100, lt=1000, title="Limit", description="Maximum number of pages to return.")
+    offset: Optional[int] = Field(default=0, title="Offset", description="Number of pages to skip.")
+
+
+class VisualizationSummary(Model):
+    id: DecodedDatabaseIdField = Field(
```

Comment:
```suggestion
    id: EncodedDatabaseIdField = Field(
```
responses should always return encoded ids